pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/bindings/LTensorTNCudaBindings.hpp

```diff
@@ -137,6 +138,20 @@ void registerBackendClassSpecificBindingsMPS(PyClass &pyclass) {
         .def(
             "updateMPSSitesData",
             [](TensorNet &tensor_network, std::vector<np_arr_c> &tensors) {
+                // Extract the incoming MPS shape
+                std::vector<std::vector<std::size_t>> MPS_shape_source;
+                for (std::size_t idx = 0; idx < tensors.size(); idx++) {
+                    py::buffer_info numpyArrayInfo = tensors[idx].request();
+                    auto MPS_site_source_shape = numpyArrayInfo.shape;
+                    std::vector<std::size_t> MPS_site_source(
+                        MPS_site_source_shape.begin(),
+                        MPS_site_source_shape.end());
+                    MPS_shape_source.push_back(std::move(MPS_site_source));
+                }
+
+                const auto &MPS_shape_dest = tensor_network.getSitesExtents();
+                MPSShapeCheck(MPS_shape_dest, MPS_shape_source);
```

Comment:
I'm not sure if this check can be inlined? I also argue this check should be checked before calling the binding or within the implementation of updating sites data.

Reply:
The initial implementation that I developed was in the Python layer before calling the binding. However, @multiphaseCFD [suggest](https://github.com/PennyLaneAI/pennylane-lightning/pull/983/#discussion_r1878564969) 
 moving this check to the cpp layer. :thinking: 
I think keeping this check on the cpp layer prevents a double computation of `sitesExtents` 

It is possible to move this check into `updateSiteData` but an early testing will prevent to compute additional steps such as `ZeroInit` or `MoveHost2Device`  which happen inside `updateSiteData`

Reply:
Agree -- thanks!